cunumeric/module.py

```diff
@@ -5327,6 +5336,630 @@ def nancumsum(
     )
 
 
+def _get_non_nan_reduction_code_if_applicable(
+    dtype: npt.DTypeLike, unary_reduction_code: UnaryRedCode
+) -> UnaryRedCode:
+    """
+    Return the equivalent non-nan reduction op codes if the datatype of the
+    array doesn't belong to one of the allowed datatypes. Raise an error if the
+    datatype is disallowed, which is currently complex64 and complex128.
+    """
+
+    equivalent_non_nan_unary_red_ops: Dict[UnaryRedCode, UnaryRedCode] = {
```

Comment:
```suggestion
    equivalent_non_nan_unary_red_ops: dict[UnaryRedCode, UnaryRedCode] = {
```
once you do this, you don't need to import `Dict` above.

Reply:
also, I'd move this to the global scope and put the name in all caps. otherwise, this dict will be created every time this function is invoked.